### PR TITLE
NXP-27488: extending ai corpus

### PR DESCRIPTION
**Pull request by andreinechaev**

### CODE REVIEW

File: nuxeo-ai-model/src/main/java/org/nuxeo/ai/adapters/AICorpus.java [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+    public void setJobId(String id) {
+        doc.setPropertyValue(CORPUS_JOBID, id);
+    }
+
+    public String getJobId() {
+        return (String) doc.getPropertyValue(CORPUS_JOBID);
+    }
+
+    public static class IOParam extends HashMap<String, String> {
```
Comment (covolution) [inline nuxeo-ai-model/src/main/java/org/nuxeo/ai/adapters/AICorpus.java]: What's this?

File: nuxeo-ai-model/src/main/java/org/nuxeo/ai/adapters/Adapter.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+ *
+ * Contributors:
+ *     anechaev
+ */
+package org.nuxeo.ai.adapters;
+
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.api.DocumentModel;
+
+public interface Adapter {
```
Comment (covolution) [inline nuxeo-ai-model/src/main/java/org/nuxeo/ai/adapters/Adapter.java]: The name and concept of an `Adapter` is already used in the Nuxeo platform.  I don't understand why you created an interface with that name.

File: nuxeo-ai-model/src/main/java/org/nuxeo/ai/model/export/DatasetExportServiceImpl.java [hunk 3]
```diff
@@ -120,22 +132,35 @@ public String export(CoreSession session, String nxql,
...
-        DocumentModel corpus = createCorpus(session, nxql, inputs, outputs, split, statsBlob);
+        AICorpus corpus = createCorpus(session, nxql, inputs, outputs, split, statsBlob);
 
         List<String> featuresList = new ArrayList<>(inputProperties);
         featuresList.addAll(outputProperties);
-        BulkCommand bulkCommand = new BulkCommand.Builder(EXPORT_ACTION_NAME, notNullNxql(nxql, featuresWithType))
+
+        String notNullNXQL = notNullNxql(nxql, featuresWithType);
+        String username = session.getPrincipal().getName();
+        BulkCommand bulkCommand = new BulkCommand.Builder(EXPORT_ACTION_NAME, notNullNXQL, username)
```
Comment (covolution) [inline nuxeo-ai-model/src/main/java/org/nuxeo/ai/model/export/DatasetExportServiceImpl.java]: Need to check whether this constructor is available on 10.10

File: nuxeo-ai-model/src/main/java/org/nuxeo/ai/model/export/DatasetExportServiceImpl.java [hunk 4]
```diff
@@ -293,4 +321,41 @@ protected String notNullNxql(String nxql, List<Map<String, String>> featuresWith
         }
         return buffy.toString();
     }
+
+    /**
+     * For the given property, find out if it exists and determine if its text or content
+     */
+    protected IOParam getPropertyWithType(String prop) {
```
Comment (covolution) [inline nuxeo-ai-model/src/main/java/org/nuxeo/ai/model/export/DatasetExportServiceImpl.java]: Why did you move these methods?

File: nuxeo-ai-model/src/main/java/org/nuxeo/ai/adapters/AICorpus.java (hunk 1 above)
Comment (andreinechaev) [reply nuxeo-ai-model/src/main/java/org/nuxeo/ai/adapters/AICorpus.java]: alias. this structure is used a lot including in other collections.

File: nuxeo-ai-model/src/main/java/org/nuxeo/ai/adapters/Adapter.java (hunk 2 above)
Comment (andreinechaev) [reply nuxeo-ai-model/src/main/java/org/nuxeo/ai/adapters/Adapter.java]: it's a default interface that allows to have consistency within the project. It requires derivatives to be able to give back document or other common props across the system.

File: nuxeo-ai-model/src/main/java/org/nuxeo/ai/model/export/DatasetExportServiceImpl.java (hunk 3 above)
Comment (andreinechaev) [reply nuxeo-ai-model/src/main/java/org/nuxeo/ai/model/export/DatasetExportServiceImpl.java]: it's not but the previous one was deprecated.

File: nuxeo-ai-model/src/main/java/org/nuxeo/ai/model/export/DatasetExportServiceImpl.java (hunk 4 above)
Comment (andreinechaev) [reply nuxeo-ai-model/src/main/java/org/nuxeo/ai/model/export/DatasetExportServiceImpl.java]: they were in another module but used in this service. I needed these helpers to be able to work with `IOParam` and to avoid cycling deps I moved them where they belong

File: nuxeo-ai-model/src/main/java/org/nuxeo/ai/adapters/AICorpus.java (hunk 1 above)
Comment (andreinechaev) [reply nuxeo-ai-model/src/main/java/org/nuxeo/ai/adapters/AICorpus.java]: maybe better naming needed

File: nuxeo-ai-model/src/main/java/org/nuxeo/ai/model/export/DatasetExportServiceImpl.java (hunk 4 above)
Comment (covolution) [reply nuxeo-ai-model/src/main/java/org/nuxeo/ai/model/export/DatasetExportServiceImpl.java]: It seems like you are more likely to be creating a cyclic dependency.  The dependencies go : pipe -> core -> model -> package.  (Model was not included for 9.10 but after all, we didn't do a release for 9.10)

Comment (covolution) [reply nuxeo-ai-model/src/main/java/org/nuxeo/ai/model/export/DatasetExportServiceImpl.java]: If you are saying its only used in 1 place and you are not duplicating code then I guess its ok to move

File: nuxeo-ai-model/src/main/java/org/nuxeo/ai/adapters/Adapter.java (hunk 2 above)
Comment (covolution) [reply nuxeo-ai-model/src/main/java/org/nuxeo/ai/adapters/Adapter.java]: Perhaps `AIAdapter` then?

File: nuxeo-ai-model/src/main/java/org/nuxeo/ai/model/export/DatasetExportServiceImpl.java (hunk 3 above)
Comment (covolution) [reply nuxeo-ai-model/src/main/java/org/nuxeo/ai/model/export/DatasetExportServiceImpl.java]: we must support 10.10

File: nuxeo-ai-model/src/main/java/org/nuxeo/ai/adapters/Adapter.java (hunk 2 above)
Comment (andreinechaev) [reply nuxeo-ai-model/src/main/java/org/nuxeo/ai/adapters/Adapter.java]: not really. It has nothing to do with AI more like a common interface that should be a common one within platform IMO

File: nuxeo-ai-model/src/main/java/org/nuxeo/ai/model/export/DatasetExportServiceImpl.java (hunk 3 above)
Comment (andreinechaev) [reply nuxeo-ai-model/src/main/java/org/nuxeo/ai/model/export/DatasetExportServiceImpl.java]: not arguing with that. It's master branch though. Aligned on non LTS Nuxeo and we use APIs that will be deprecated at some point. Just wanted to avoid that. Moving this part to 10.10 will requires some changes but at some point this difference will be there. So why not do it now?

Comment (covolution) [reply nuxeo-ai-model/src/main/java/org/nuxeo/ai/model/export/DatasetExportServiceImpl.java]: AI is an addon so it doesn't necessaryily follow the versioning of the platform.  I am trying to keep our `master` branch compatible with 10.10

File: nuxeo-ai-model/src/main/java/org/nuxeo/ai/adapters/Adapter.java (hunk 2 above)
Comment (covolution) [reply nuxeo-ai-model/src/main/java/org/nuxeo/ai/adapters/Adapter.java]: That's my point.  Its already a concept in the platform but you are adding it in the AI module

Comment (andreinechaev) [reply nuxeo-ai-model/src/main/java/org/nuxeo/ai/adapters/Adapter.java]: renamed to `AIAdapter`

File: nuxeo-ai-model/src/main/java/org/nuxeo/ai/model/export/DatasetExportServiceImpl.java (hunk 3 above)
Comment (andreinechaev) [reply nuxeo-ai-model/src/main/java/org/nuxeo/ai/model/export/DatasetExportServiceImpl.java]: moved back to deprecated constructor
